clang/lib/CodeGen/CGExprCXX.cpp

```diff
@@ -2216,7 +2216,7 @@ static llvm::Value *EmitTypeidFromVTable(CodeGenFunction &CGF, const Expr *E,
 }
 
 llvm::Value *CodeGenFunction::EmitCXXTypeidExpr(const CXXTypeidExpr *E) {
-  llvm::Type *PtrTy = llvm::PointerType::getUnqual(getLLVMContext());
+  llvm::Type *PtrTy = Int8PtrTy;
```

Comment:
Should this be `GlobalsInt8PtrTy`?

Reply:
It should but sadly it cannot, see our historical conversation here: <https://reviews.llvm.org/D157452>. I've not got around to working on your suggestion there about supporting declaring a default AS for a class, so we have to keep things like so for now.

Reply:
I also find this somewhat surprising. Looking at the discussion that linked review the concern seems to be that you'd end up emitting e.g. `ptr addrspace(1)` here and users of EmitCXXTypeidLValue do not expect to handle that?

Could you add a comment that the result of this type is used in contexts where the "default" address space is expected. Without the comment I find this very confusing.

Reply:
Done-ish, please let me know if this is not actually helpful in clarifying and I can refine / reword.

Reply:
TODO seems like a strange way to phrase this if it's true that this can never be GlobalsInt8PtrTy

Reply:
It could be `GlobalsInt8PtrTy`, but the main roadblock would require something that @rjmccall suggested looking into, namely adding the ability to declare a default AS for a class type, which stdlib implementations could then re-use. Having said that, I've neither had the time to look into it, nor figured out if this would work in general, considering we need to compose with stdlib implementations other than libc++. Having said that, perhaps the TODO: should actually be at the end of the comment, and just say something along the lines of "investigate if it is possible to remove this limitation"?